mobx_codegen/lib/src/template/store.dart

```diff
@@ -1,12 +1,18 @@
 import 'package:mobx_codegen/src/template/action.dart';
 import 'package:mobx_codegen/src/template/async_action.dart';
+import 'package:mobx_codegen/src/template/comma_list.dart';
 import 'package:mobx_codegen/src/template/computed.dart';
 import 'package:mobx_codegen/src/template/observable.dart';
 import 'package:mobx_codegen/src/template/observable_future.dart';
 import 'package:mobx_codegen/src/template/observable_stream.dart';
+import 'package:mobx_codegen/src/template/params.dart';
 import 'package:mobx_codegen/src/template/rows.dart';
 
 class StoreTemplate {
+  final SurroundedCommaList<TypeParamTemplate> typeParams =
```

Comment:
Might also need to check for type-parameters for fields, eg: `ObservableList<T>` where the `T` comes from the class type param.

Reply:
The PR just changes the way codegen detects the class the mixin is applied to, and adds generic params to the mixin definition. Class body is unaffected, and using the generic params in fields etc. Just Works™